presto-main/src/test/java/com/facebook/presto/sql/gen/TestPageFunctionCompiler.java

```diff
@@ -116,14 +117,17 @@ public void testGeneratedClassName()
     {
         PageFunctionCompiler functionCompiler = new PageFunctionCompiler(createTestMetadataManager(), 0);
 
+        String dateTimePattern = "YYYYMMdd_HHmmss_SSS";
         String planNodeId = "7";
         String stageId = "20170707_223500_67496_zguwn.2";
         String classSuffix = stageId + "_" + planNodeId;
         Supplier<PageProjection> projectionSupplier = functionCompiler.compileProjection(ADD_10_EXPRESSION, Optional.of(classSuffix));
         PageProjection projection = projectionSupplier.get();
         Work<Block> work = projection.project(SESSION, new DriverYieldSignal(), createLongBlockPage(0), SelectedPositions.positionsRange(0, 1));
         // class name should look like PageProjectionOutput_20170707_223500_67496_zguwn_2_7_XX
-        assertTrue(work.getClass().getSimpleName().startsWith("PageProjectionWork_" + stageId.replace('.', '_') + "_" + planNodeId));
+        String className = work.getClass().getSimpleName();
+        assertTrue(className.startsWith("PageProjectionWork_" + stageId.replace('.', '_') + "_" + planNodeId));
+        DateTimeFormat.forPattern(dateTimePattern).parseDateTime(className.substring(className.length() - dateTimePattern.length()));
```

Comment:
What does this check? The parse succeed without throwing an exception? 

Can we do something like parse the date time and assert it's <= current timestmap? @haozhun

Reply:
Also, assert there is nothing between `PageProjectionOutput_20170707_223500_67496_zguwn_2_7_XX` and the timestamp.

Reply:
TBH, I wouldn't even test for this. It's an implementation detail, not a contract for how the generator works.

Reply:
When you verify this manually, make sure these timestamps are generated in the same zone as the timestamps in our query id, which is also commonly seen in our stack traces.

Reply:
In `QueryIdGenerator` we are also using UTC timezone to generate timestamp ([here](https://github.com/prestodb/presto/blob/c73359fe2173e01140b7d5f102b286e81c1ae4a8/presto-main/src/main/java/com/facebook/presto/execution/QueryIdGenerator.java#L46)). So we should be fine.